core/src/wsv.rs

```diff
@@ -577,14 +569,12 @@ impl WorldStateView {
     }
 
     /// Returns iterator over blockchain blocks starting with the block of the given `height`
-    pub fn blocks_from_height(
-        &self,
-        height: usize,
-    ) -> impl Iterator<Item = VersionedCommittedBlock> + '_ {
+    pub fn blocks_from_height(&self, height: usize) -> Vec<VersionedCommittedBlock> {
```

Comment:
`impl Iterator` seems better. You can postpone collecting if you need to.

Reply:
This can be explored in the future. I made this choice to avoid contention between threads since the wsv is currently very global.